### ISSUE TITLE
Named mutex creation not working in some docker images

### ISSUE THREAD

**Issue by kouvel** [dotnet/coreclr#17098]
Forked from https://github.com/dotnet/roslyn/issues/24124

Info so far:
- We don't yet have a repeatable set of repro steps
- /tmp appears to be mapped to the host in some unusual way. It is not clear yet what is remapping it, if it is mapped already in the base image, if that is a standard thing to do, or if that is the cause of the problem
- Named mutex creation fails (see logs in original issue above)
- Only happens with recent SDK (2.2.0-preview1-007866 and later)
- Dotnet is installed with dotnet-install.sh script
- It is not clear which base docker image is used

@sebastienros @jaredpar

**Comment by kouvel** [comment]
Named mutexes work as expected on the Microsoft-published 2.0.6-runtime docker image, need to try with SDK above and using the script.

**Comment by kouvel** [comment]
Some pending questions:
- Which base docker image is being used? A repeatable set of repro steps would be preferable, starting with the Dockerfile.
- What is the output of df -h /tmp?
- Is there some setup step that is remapping tmp or is that already done in the base image?

**Comment by sebastienros** [comment]
```
root@asp-perf-lin:/benchmarks#  df -h /tmp
Filesystem                           Size  Used Avail Use% Mounted on
/dev/mapper/asp--perf--lin--vg-root  203G  140G   53G  73% /tmp
```

Docker file:
https://github.com/aspnet/benchmarks/blob/dev/docker/benchmarks/Dockerfile

Started using this command:
https://github.com/aspnet/benchmarks/blob/dev/docker/benchmarks/run-server.sh#L78

Then dotnet is installed at runtime using `dotnet-install` and the application that we try to build using it is https://github.com/aspnet/benchmarks/tree/dev/src/Benchmarks

I can also give you access to the machine to run inside the docker and repro the issue for you. I am in building 18.

**Comment by qmfrederik** [comment]
@kouvel I'm also hitting this problem.

* Base docker image: https://github.com/quamotion/docker-build/blob/702149e543712a185c76bd9af8ee67f61e3ec835/gitlab-linux/Dockerfile, or `docker pull quay.io/quamotion/gitlab-runner-linux:10.3.0-u1`
* Not sure, need to run
* Not in the base image as far as I know, but this runs as a GitLab Docker agent; not sure whether they remap `/tmp` or not.

**Comment by qmfrederik** [comment]
Also want to add that the docker images have access to 32 cores, not sure if that's relevant. We migrated to 32 cores about at the same time when we started seeing this issue, not sure if it's related or not.

**Comment by mikeharder** [comment]
@kouvel: I was also able to repro this twice today, but I can't get it to repro again.  The app I was compiling was under `/tmp`, but `/tmp` is not remapped:

```
# df -h /tmp
Filesystem      Size  Used Avail Use% Mounted on
overlay          59G   12G   45G  20% /
```

Here's the dockerfile I used:

```
FROM buildpack-deps:stretch-scm

# Install .NET CLI dependencies
RUN apt-get update \
    && apt-get install -y --no-install-recommends \
        libc6 \
        libcurl3 \
        libgcc1 \
        libgssapi-krb5-2 \
        libicu57 \
        liblttng-ust0 \
        libssl1.0.2 \
        libstdc++6 \
        libunwind8 \
        zlib1g \
    && rm -rf /var/lib/apt/lists/*

# Install .NET Core SDK
ENV DOTNET_SDK_VERSION 2.1.300-preview3-008387

RUN curl -SL --output dotnet.tar.gz https://dotnetcli.blob.core.windows.net/dotnet/Sdk/$DOTNET_SDK_VERSION/dotnet-sdk-$DOTNET_SDK_VERSION-linux-x64.tar.gz \
    && mkdir -p /usr/share/dotnet \
    && tar -zxf dotnet.tar.gz -C /usr/share/dotnet \
    && rm dotnet.tar.gz \
    && ln -s /usr/share/dotnet/dotnet /usr/bin/dotnet

ENV NUGET_XMLDOC_MODE skip

# Trigger first run experience by running arbitrary cmd to populate local package cache
RUN dotnet help
```

**Comment by kouvel** [comment]
@sebastienros, I think the issue may be that the --mount command in the .sh file above seems to mount /tmp in the container with the same permissions as the source /mnt, and coreclr currently requires 777 permissions on the folder. Those permissions are necessary for everything under /tmp/.dotnet/shm due to global mutexes, the requirement on /tmp and /tmp/.dotnet shouldn't be necessary. Can you try to `chmod 777 /mnt` first and see if it works after? If you can confirm that's the issue then you can use that as a workaround for now and I'll put in a fix.

@qmfrederik, are you seeing the issue consistently? If so, can you confirm if it's the same issue with /tmp remapping?

@mikeharder that seems like a different issue from the one @sebastienros is seeing. Could you please provide repro steps for building the project?

**Comment by kouvel** [comment]
Closing due to lack of further information

**Comment by sebastienros** [comment]
Sorry, I was almost sure I had answered your comment, `chmod 777 /mnt` fixes the issue.

**Comment by kouvel** [comment]
Great, thanks! The fix has been checked in for 2.1.
